tools/lib/template_parser.py

```diff
@@ -236,6 +236,26 @@ def is_django_block_tag(tag):
         'raw',
     ]
 
+def extract_line(text, offset):
+    # type: (str, int) -> Dict[str, Any]
+    def line_number_of_offset(offset, lookup):
+        # type: (int, str) -> int
+        line_count = 1
+        for line in text.split('\n'):
+            if lookup in line:
+                break
+            line_count += 1
+        return line_count
+
+    details = {} # type: Dict[str, Any]
+    end_of_line = offset
+    while text[end_of_line] != '\n':
+        end_of_line += 1
```

Comment:
I believe we do, unless there's a simpler way of determining the end of the line. It's used here: `details['lookup'] = text[offset:end_of_line]`.